### DISCUSSION TITLE
Unable to add custom labels to spacy doc spans

### DISCUSSION THREAD

**Discussion by ayoub-chammam** [explosion/spaCy#11251] [Help: Coding & Implementations] [answered]
Hi, I am trying to insert NER labels into Spacy doc,
my data is like this: ("some sentence text", [(0, 5, "label")])
I want to insert this into spans and be able to also set it as default when calling doc.ents , any ideas how ?

**Comment by ayoub-chammam** [comment, marked as answer]
I've been looking for an answer for quite some time and I finally figured out how to apply this (turned out to be so simple) so I'm posting my approach here so maybe other people could share their insights:
```
import spacy
text = 'EU rejects German call to boycott British lamb.'
nlp = spacy.load('en_core_web_sm')
doc = nlp(text)
offsets = [(0, 2, 'ORG'), (11, 17, 'MISC'), (34, 41, 'MISC')]
spans = [doc.char_span(x[0], x[1], label=x[2]) for x in offsets]
doc.spans['gold'] = spans
```
- another approach I find was to apply only the Tokenizer from SpaCy:

```
tokens = tokenizer(text)
words = [token.text for token in tokens]
doc = Doc(Vocab(), words)
```

**Reply by polm** [reply]
Glad you figured it out! For the record an example of handling this data is in the [training data docs](https://spacy.io/usage/training#training-data).

**Reply by ayoub-chammam** [reply]
How did I miss this! Thank you!!  :smiley:
